src/CalcViewModel/Common/KeyboardShortcutManager.cpp

```diff
@@ -83,31 +83,15 @@ namespace CalculatorApp
             // The button will go into the visual Pressed state with this call
             VisualStateManager::GoToState(button, "Pressed", true);
 
-            // This timer will fire after lightUpTime and make the button
-            // go back to the normal state.
-            // This timer will only fire once after which it will be destroyed
-            auto timer = ref new DispatcherTimer();
-            TimeSpan lightUpTime{};
-            lightUpTime.Duration = 500000L; // Half second (in 100-ns units)
-            timer->Interval = lightUpTime;
-
-            WeakReference timerWeakReference(timer);
-            WeakReference buttonWeakReference(button);
-            timer->Tick += ref new EventHandler<Object ^>([buttonWeakReference, timerWeakReference](Object ^, Object ^) {
-                auto button = buttonWeakReference.Resolve<ButtonBase>();
-                if (button)
-                {
-                    VisualStateManager::GoToState(button, "Normal", true);
-                }
+            winrt::apartment_context uiThreadContext;
 
-                // Cancel the timer after we're done so it only fires once
-                auto timer = timerWeakReference.Resolve<DispatcherTimer>();
-                if (timer)
-                {
-                    timer->Stop();
-                }
-            });
-            timer->Start();
+            co_await winrt::resume_background();
+            co_await winrt::resume_after(500ms);
```

Comment:
Not related to this PR, but we should consider reverting this change or altering the ms here. To me the keypad feels laggier with this change, but I could see the argument that it makes it clearer visually which key was activated.